tracee-ebpf/tracee/events_pipeline.go

```diff
@@ -39,25 +40,39 @@ type context struct {
 	_        [3]byte //padding
 }
 
-func (t *Tracee) processEvents(done <-chan struct{}) error {
+// handleEvents is a high-level function that starts all operations related to events processing
+func (t *Tracee) handleEvents(done <-chan struct{}) {
+	ctx, cancelFunc := context.WithCancel(context.Background())
```

Comment:
why another cancellation signal in addition to `done`?

Reply:
if we would have used done, we would have to send done signals as the number of pipeline stages (as once one of the stages consumes the event, another one won't get it).
Using context resolves this, as this is what it is intended to do - to signal cancellation.
In a future PR, I want to even move the context into main, and remove the done channel.

Reply:
the channel is not intended to be written to. It is intended to be `close`d when finished (this is a common go pattern). https://github.com/aquasecurity/tracee/blob/main/tracee-ebpf/tracee/tracee.go#L979
close is broadcasted to all consumers

Reply:
It's even discussed in the blog post you referred to:

> This approach has a problem: each downstream receiver needs to know the number of potentially blocked upstream senders and arrange to signal those senders on early return. Keeping track of these counts is tedious and error-prone.
We need a way to tell an unknown and unbounded number of goroutines to stop sending their values downstream. In Go, we can do this by closing a channel, because a receive operation on a closed channel can always proceed immediately, yielding the element type’s zero value.
This means that main can unblock all the senders simply by closing the done channel. This close is effectively a broadcast signal to the senders. We extend each of our pipeline functions to accept done as a parameter and arrange for the close to happen via a defer statement, so that all return paths from main will signal the pipeline stages to exit.

Reply:
Fair enough. I'll remove the context part from this PR and will add it in a future PR when it will be also part of main (also related to #975)